### PR TITLE
Prepend JMSDiExtraBundle configuration making FosRestBundle and NelmioApiDocBundle optional

### LINKED ISSUE #258: Add missing dependency "nelmio/api-doc-bundle" to composer.json
Hi there,

since the controllers (e.g. Api/CategoryController) using the ApiDoc-annotation from the Nelmio\ApiDocBundle, this dependency should be consider in the composer.json

### PR DESCRIPTION
**Pull request by gremo**
I am targetting this branch, because this PR make possible to use JMSDiExtra bundle without FosRestBundle  and NelmioApiDocBundle.

Sorry for linking another bundle issue but it's exactly the same problem with the same solution (I suppose): https://github.com/sonata-project/SonataUserBundle/issues/686

Closes #258

## Changelog

```markdown
### Added
- Prepended JMSDiExtraBundle configuration
```

### CODE REVIEW

**Comment by gremo** [comment]
Woops! Sorry for using the `[]` notation...

**Comment by greg0ire** [comment]
It looks like all these warnings are suddenly an issue :/

**Comment by gremo** [comment]
@greg0ire is there something I can do?

**Comment by greg0ire** [comment]
See https://github.com/sonata-project/SonataAdminBundle/pull/4288, any help would be very welcome!

**Comment by gremo** [comment]
Ok @greg0ire I'll take a look. In the meanwhile you agree that this PR should be merged even if Travis complains?

**Comment by greg0ire** [comment]
Should, yes, can, no… the merge button is disabled as soon as one of the checks marked as "Required" is KO (which is a very good thing, trust me).

**Comment by gremo** [comment]
OK then, I'll take a look this evening. I don't understand if these warnings are my bad or is something that doesn't depend on the PR itself.

**Comment by greg0ire** [comment]
They are definitely not your fault ;)

**Comment by greg0ire** [comment]
Restarting the build, the bug was fixed.

**Comment by gremo** [comment]
Good! @greg0ire what do you think about @kaiwa comment in https://github.com/sonata-project/SonataUserBundle/issues/686?

File: DependencyInjection/SonataClassificationExtension.php [hunk 1]
```diff
@@ -239,4 +240,30 @@ public function registerDoctrineMapping(array $config)
...
+    /**
+     * {@inheritdoc}
+     */
+    public function prepend(ContainerBuilder $container)
+    {
+        $bundles = $container->getParameter('kernel.bundles');
+
+        // Blacklist all API controllers when using JMSDiExtraBundle without NelmioApiDocBundle or FOSRestBundle. API
+        // controllers use annotations and would be warmed, throwing an exception.
+        if (isset($bundles['JMSDiExtraBundle']) &&
```
Comment (core23) [inline DependencyInjection/SonataClassificationExtension.php]: Please combine the `isset`s @gremo

Comment (gremo) [inline DependencyInjection/SonataClassificationExtension.php]: @core23 done!

**Comment by greg0ire** [comment]
> Good! @greg0ire what do you think about @kaiwa comment in sonata-project/SonataUserBundle#686?

@gremo which one? Please link to the exact comment.

**Comment by gremo** [comment]
@greg0ire [this](https://github.com/sonata-project/SonataUserBundle/issues/686#issuecomment-275370454) comment. Are we sure we want to merge this instead of forcing the aproval of https://github.com/schmittjoh/JMSDiExtraBundle/pull/270?

@OskarStark I'll try, but I'm not good at writing automatic tests 😢

**Comment by greg0ire** [comment]
schmittjoh/JMSDiExtraBundle#270 might get merged. I found some tiny issues that you will be able to fix quickly.

**Comment by gremo** [comment]
You are an hero!!! As soon I'll come back home I'll fix those issues... thanks :)

**Comment by gremo** [comment]
So @greg0ire I think we can close this thanks to https://github.com/schmittjoh/JMSDiExtraBundle/pull/270?

**Comment by greg0ire** [comment]
A PR to require the version that has the fix might be in order though

**Pull request closed without merging**